extensions/positron-python/python_files/positron/positron_ipykernel/connections.py

```diff
@@ -6,7 +6,17 @@
 
 import logging
 import uuid
-from typing import TYPE_CHECKING, Any, Dict, List, Optional, Set, Tuple, TypedDict, Union
+from typing import (
```

Comment:
Do you know if your current formatting setup could be configured to avoid changes like this?

IIUC this change was not required by ruff, which is what we use in CI, otherwise we would have seen this as a formatting fix or error already. However, once you do format this over multiple lines (and if you have a trailing comma on the last line), ruff will respect that.

We can merge this PR as is to keep things moving, but I think this is worth resolving to reduce diff sizes moving forward. For example, most of the changes in this file are related to the inconsistency I described.

Reply:
This seems to be a tug of war between black in VSCode and Ruff that has been going on a while — @dfalbel couldn’t figure out why it was doing this. Is there a way to force Ruff to reformat to remove line breaks? When I load files in emacs, I format them to 79 characters so the files are readable and then they are reformatted to 99

Reply:
Personally I would rather we used 79 characters instead of 99.

Reply:
Let's adhere to the settings we have now. Upstream ruff config also has 100 line-length - this seems reasonable to me.

Reply:
If we stick to 100 characters, I do think we need to address this issue. I don't think it's a tug of war since you can reproduce the issue with Ruff only. Formatting with 79 characters and then 100 characters doesn't necessarily restore the original document.

It's described a bit here: https://docs.astral.sh/ruff/settings/#format_skip-magic-trailing-comma. We can't just enable that setting though, since that'll format a bunch of upstream files. If we could _only_ apply that setting to Positron source files it would allow @wesm to use his workflow while the committed files stay at 100 characters.

Reply:
I’m a bit of a PEP8 dogmatist (https://peps.python.org/pep-0008/#maximum-line-length). If we can change the Ruff settings to be deterministic without disrupting the upstream files it seems like that would be ideal. I can also just use that setting in my local setup.